src/Words.y

```diff
@@ -13,116 +13,401 @@
+	|	MAKE reference_list identifier SAY value_expression now '.'					{ $$ = new INode(AST.Type.QUEUE_SAY, $2, $3, $5, $6); }
+	|	MAKE reference_list identifier WAIT value_expression TURNS '.'				{ $$ = new INode(AST.Type.QUEUE_WAIT, $2, $3, $5, null); }
+	|	MAKE reference_list identifier WAIT value_expression TURNS now '.'			{ $$ = new INode(AST.Type.QUEUE_WAIT, $2, $3, $5, $7); }
+	|	STOP reference_list identifier '.'											{ $$ = new INode(AST.Type.QUEUE_STOP, $2, $3); }
+	|	queueing_custom_action_statement											{ $$ = $1; }
+	;
+
+queueing_custom_action_statement:
+		MAKE reference_list identifier identifier '.'								{ $$ = new INode(AST.Type.QUEUE_ACTION, $2, $3, $4, null, null); }
+	|	MAKE reference_list identifier identifier now '.'							{ $$ = new INode(AST.Type.QUEUE_ACTION, $2, $3, $4, null, $5); }
+	|	MAKE reference_list identifier identifier WITH parameter_list '.'			{ $$ = new INode(AST.Type.QUEUE_ACTION, $2, $3, $4, $6, null); }
+	|	MAKE reference_list identifier identifier WITH parameter_list now '.'		{ $$ = new INode(AST.Type.QUEUE_ACTION, $2, $3, $4, $6, $7); }
+	;
+
+predicate:
+		basic_action_predicate														{ $$ = $1; }
+	|	boolean_predicate															{ $$ = $1; }
+	;
+
+basic_action_predicate:
+		reference_list identifier MOVES												{ $$ = new INode(AST.Type.MOVES_PREDICATE, $1, $2, null); }
+	|	reference_list identifier MOVES direction									{ $$ = new INode(AST.Type.MOVES_PREDICATE, $1, $2, $4); }
+	|	reference_list identifier SAYS value_expression								{ $$ = new INode(AST.Type.SAYS_PREDICATE, $1, $2, $4); }
+	|	reference_list identifier WAITS												{ $$ = new INode(AST.Type.WAITS_PREDICATE, $1, $2); }
+	;
+
+boolean_predicate:
+		relational_expression						{ $$ = $1; }
+	|	'(' boolean_predicate ')'					{ $$ = $2; }
+	|	NOT '(' boolean_predicate ')'				{ $$ = new INode(AST.Type.NOT, $3); }
+	|	boolean_predicate AND boolean_predicate		{ $$ = new INode(AST.Type.AND, $1, $3); }
+	|	boolean_predicate OR boolean_predicate		{ $$ = new INode(AST.Type.OR, $1, $3); }
+	;
+
+relational_expression:
+		value_expression '=' value_expression		{ $$ = new INode(AST.Type.EQUALS, $1, $3); }
+	|	value_expression '<' value_expression		{ $$ = new INode(AST.Type.LESS, $1, $3); }
+	|	value_expression '>' value_expression		{ $$ = new INode(AST.Type.GREATER, $1, $3); }
+	|	value_expression LEQ value_expression		{ $$ = new INode(AST.Type.LEQ, $1, $3); }
+	|	value_expression GEQ value_expression		{ $$ = new INode(AST.Type.GEQ, $1, $3); }
+	;
+
+value_expression:
+		reference_list identifier					{ $$ = new INode(AST.Type.RETRIEVE_PROPERTY, $1, $2); }
+	|	literal										{ $$ = $1; }
+	|	NOTHING										{ $$ = new LNode(AST.Type.NOTHING); }
+	|	'(' value_expression ')'					{ $$ = $2; }
+	|	'-' value_expression %prec UMINUS			{ $$ = new INode(AST.Type.NEGATE, $2); }
+	|	value_expression '+' value_expression		{ $$ = new INode(AST.Type.ADD, $1, $3); }
```

Comment:
The reason to do the arithmetic in the AST and not Yacc is type checking and coercion.  Since we have dynamic typing and the identifier could refer to anything (number, string, object), we have to build the AST and do the operand type checking (and coercion) during evaluation.

I also think it's much cleaner this way, as it provides a true separation of concerns.  The parser builds the AST, but it has no role in evaluation.